lib/new_relic/agent/instrumentation/grpc/client/instrumentation.rb

```diff
@@ -13,22 +13,41 @@ module GRPC
         module Client
           include NewRelic::Agent::Instrumentation::GRPC::Helper
 
-          def issue_request_with_tracing(method, requests, marshal, unmarshal,
+          # TODO: gRPC - record request type
+          def issue_request_with_tracing(grpc_type, method, requests, marshal, unmarshal,
             deadline:, return_op:, parent:, credentials:, metadata:)
             return yield unless trace_with_newrelic?
 
             segment = request_segment(method)
             request_wrapper = NewRelic::Agent::Instrumentation::GRPC::Client::RequestWrapper.new(@host)
             segment.add_request_headers(request_wrapper)
             metadata.merge!(metadata, request_wrapper.instance_variable_get(:@newrelic_metadata))
+            grpc_message = nil
+            grpc_status = 0
 
             NewRelic::Agent.disable_all_tracing do
-              NewRelic::Agent::Tracer.capture_segment_error(segment) do
+              begin
                 yield
+              rescue => e
+                NewRelic::Agent.notice_error(e)
+                if e.message =~ /debug_error_string:(.*)$/
+                  hash = JSON.parse(Regexp.last_match(1))
+                  grpc_message = hash['grpc_message']
+                  grpc_status = hash['grpc_status']
```

Comment:
Is `debug_error_string` JSON? Should that top-level key be parsed instead?

Reply:
Addressed with a389f1de67f742ca97d941906cd5dbe5fc2627ae

The message text is plain text with a JSON blob inside of it. 😞 Seeing as we have to take a regex parse hit, let's just grab the code and status with regex.